### DISCUSSION TITLE
How to install 11ty v1.0 new beta version?

### DISCUSSION THREAD

**Discussion by cekvenich2** [11ty/eleventy#2030] [Uncategorized]
How to install 11ty v1.0 new beta version?

Also - should I?

**Comment by cfjedimaster** [comment]
Per the blog post (https://www.11ty.dev/blog/eleventy-v1-beta/), you can do:

```
# Local project
npm install @11ty/eleventy@beta

# Global install
npm install @11ty/eleventy@beta -g
```

Should you. Well... that depends. :)  There are going to be some backwards compat issues you may run into. I doscumented my process here (https://www.raymondcamden.com/2021/10/08/eleventy-10-upgrading-experience) and in my huge site (6k+ files), it was mostly painless. I think the biggest issues for me were invalid Liquid filters and non-quoted Liquid includes.

My site is just a personal blog so it was a safe gamble. You need to decide for yourself how risky the move would be. But in general, I think it's relatively safe. And heck. you can roll back in a minute.

**Reply by desidem** [reply]
[I hope this is appropriate to post here. Should I post elsewhere?]

If I wanted to 'roll back' to my previous version of 11ty how would I do that? Through Github? Or would I uninstall/reinstall 11ty? Perhaps both Github and un-install/reinstall?

1.0.0 was just released. I 'accidentally' updated my site, which was previously a local install of the last 11ty version, because I thought it would be best. And then my site wouldn't build. I fixed the issue (I think) but in case there's problems ahead I'd rather be prepared.

**Reply by pdehaan** [reply]
@desidem What do you mean by "previous version"?
You want 0.12.1? Or an earlier 1.0.0-beta.x build?

Personally I usually just delete the ./node_modules/ directory and package-lock.json file and then run something like <kbd>npm i @11ty/eleventy@0.12 -D</kbd> to reinstall the latest 0.12.x version and save it as a devDependency.

**Comment by cekvenich2** [comment]
Thank you! (Also a big plus for 11ty is a great user community. )

**Comment by desidem** [comment]
It was 0.12.1. TI started to panic when my build failed. Fixing it is one
thing. Being able to undo it is another. Thank you so much for these
instructions!

**Reply by pdehaan** [reply]
If you have any other questions, probably easier to start a new discussion/issue to avoid too many concurrent conversations in a closed discussion thread.

If your repo is public, I can clone it locally and we can troubleshoot why it works in 0.12.x but not in 1.0.x.
I think there was an issue w/ `pagination` in 1.0.0 when not explicitly setting `permalink`.

@zachleat also published a 1.0.1-canary.2 build (on 2022-01-13T16:25:10.159Z; <kbd>npm install @11ty/eleventy@canary -D</kbd>) which might have fixed whatever issues you were seeing w/ 1.0.0.

```sh
npm info @11ty/eleventy dist-tags versions --json

{
  "dist-tags": {
    "latest": "1.0.0",
    "beta": "1.0.0-beta.10",
    "canary": "1.0.1-canary.2"
  },
  "versions": [
    ...
    "0.12.1",
    "1.0.0-beta.1",
    ...
    "1.0.0-beta.10",
    "1.0.0-canary.1",
    ...
    "1.0.0-canary.50",
    "1.0.0",
    "1.0.1-canary.1",
    "1.0.1-canary.2"
  ]
}
```

**Comment by desidem** [comment]
I wasn't sure if asking about roll-back was worthy of separate discussion.
But next time I'll know better.
I know what the build problem (in general) was:

Error logs flagged it as a Liquid problem. Long story short: I have an
email template that utilizes Go variables because it's a custom
confirmation email for Netlify (variables for the site name, tokens, etc).
In order to get that working originally (in 0.12.1), I had to use
passthrough in the eleventy.js file (it was a bit of a struggle to get it
working).  Now,with 11ty 1.0.0 it seems to be an issue even with
passthroughs in the eleventy.js file. I created an .eleventyignore file to
ignore the email templates. It's working and the site has built, but I was
worried there may be something lurking and wanted to know how to roll back.

Should I post my error log? I realize now that you might *want* to know
about these issues. Apologies in advance if not following protocol. If you
require me to post elsewhere let me know and I'll start a new thread.
